### PR TITLE
Feature/refactor corona test service

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Introduces a protocol and mock implementation for the `CoronaTestService` to simplify its use in unit tests.

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/__tests__/HomeTestResultCellModelTests.swift [hunk 1]
```diff
@@ -58,7 +58,6 @@ class HomeTestResultCellModelTests: CWATestCase {
 		AccessibilityIdentifiers.Home.TestResultCell.loadingPCRButton
 	]
 
-	// swiftlint:disable:next function_body_length
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/TestResult/__tests__/HomeTestResultCellModelTests.swift:61]: :D

**Review by marcussc — APPROVED** [review]
The amount of change was scary but then it was quite pleasant to review

File: src/xcode/ENA/ENA/Source/Scenes/Home/__tests__/HomeTableViewModelTests.swift [hunk 2]
```diff
@@ -286,29 +191,9 @@ class HomeTableViewModelTests: CWATestCase {
...
-				dscListProvider: MockDSCListProvider(),
-				client: client,
-				appConfiguration: appConfiguration,
-				cclService: FakeCCLService(),
-				recycleBin: .fake()
-			),
-			recycleBin: .fake(),
-			badgeWrapper: badgeWrapper
-		)
+		let coronaTestService = MockCoronaTestService()
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/__tests__/HomeTableViewModelTests.swift:196]: ❤️

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestServiceProviding.swift [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+		lastName: String?,
+		dateOfBirth: String?,
+		isSubmissionConsentGiven: Bool,
+		markAsUnseen: Bool,
+		certificateSupportedByPointOfCare: Bool,
+		certificateConsent: TestCertificateConsent,
+		completion: @escaping TestResultHandler
+	)
+	
+	func reregister(coronaTest: CoronaTest)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestServiceProviding.swift:82]:
```suggestion
	func reRegister(coronaTest: CoronaTest)
```

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/MockCoronaTestService.swift [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+		markAsUnseen: Bool,
+		certificateSupportedByPointOfCare: Bool,
+		certificateConsent: TestCertificateConsent,
+		completion: @escaping TestResultHandler
+	) {
+		onRegisterRapidPCRTestAndGetResult()
+		completion(registerRapidPCRTestAndGetResultResult ?? .failure(.noCoronaTestOfRequestedType))
+	}
+	
+	func reregister(coronaTest: CoronaTest) {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/__tests__/MockCoronaTestService.swift:107]:
```suggestion
	func reRegister(coronaTest: CoronaTest) {
```

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**